crates/dekaf/src/session.rs

```diff
@@ -1142,48 +1143,68 @@ impl Session {
+                    tracing::warn!(
+                        topic = decrypted_name.as_str(),
+                        partition = partition.partition_index,
+                        ?error,
+                        "Got error from upstream Kafka when trying to commit offsets"
+                    );
                 } else {
+                    let response_partition_index = partition.partition_index;
+
                     let journal_name = collection_partitions
-                        .get(partition.partition_index as usize)
+                        .get(response_partition_index as usize)
                         .context(format!(
-                            "unable to find partition {} in collection {:?}",
-                            partition.partition_index, topic.name
+                            "unable to find collection partition idx {} in collection {:?}",
+                            response_partition_index,
+                            decrypted_name.as_str()
                         ))?
                         .spec
                         .name
                         .to_owned();
 
-                    let committed_offset = req
+                    let request_partitions = &req
                         .topics
                         .iter()
-                        .find(|req_topic| req_topic.name == topic.name)
-                        .context(format!("unable to find topic in request {:?}", topic.name))?
-                        .partitions
-                        .get(partition.partition_index as usize)
+                        .find(|req_topic| req_topic.name == encrypted_name)
+                        .context(format!(
+                            "unable to find topic in request {:?}",
+                            decrypted_name.as_str()
+                        ))?
+                        .partitions;
+
+                    let committed_offset = request_partitions
+                        .iter()
+                        .find(|req_part| req_part.partition_index == response_partition_index)
                         .context(format!(
-                            "unable to find partition {}",
-                            partition.partition_index
+                            "Unable to find partition index {} in request partitions for topic {:?}, though response contained it. Request partitions: {:?}. Flow has: {:?}",
+                            response_partition_index,
+                            decrypted_name.as_str(),
+                            request_partitions,
+                            collection_partitions
                         ))?
                         .committed_offset;
 
                     metrics::gauge!("dekaf_committed_offset", "group_id"=>req.group_id.to_string(),"journal_name"=>journal_name).set(committed_offset as f64);
                 }
             }
+
+            tracing::info!(topic_name = ?topic.name, partitions = ?topic.partitions, "Committed offset");
```

Comment:
Would it be helpful to include the actual offsets in this log, maybe just as an array?

Reply:
I moved the log so it logs per partition -- I don't think this will make logs any noisier, and it's arguably easier to read than parsing an array out of a single log line. Most of the journals have a single partition anyway